pkg/lib/infra/sms/config_resolver.go

```diff
@@ -0,0 +1,118 @@
+package sms
+
+import (
+	"fmt"
+
+	"github.com/authgear/authgear-server/pkg/lib/config"
+)
+
+type ConfigResolver struct {
+	AuthgearSecretsYAMLSMSProvider config.SMSProvider
+	AuthgearSecretsYAMLSMSGateway  *config.SMSGateway
```

Comment:
The name is confusing. Do you mean `AuthgearYAMLSMSProvider`, `AuthgearYAMLSMSGateway`?